File: Leetcode/cpp/sol3498.cpp
# https://leetcode.com/problems/reverse-degree-of-a-string/

class Solution {
public:
    int reverseDegree(string s) {
        int result = 0;
        for(auto i = 0;i<s.size();i++){
            int ord = s[i] - 96;
            result = result + ((27-ord)*(i+1));
        }
        return result;
    }
};

File: Leetcode/python/sol3498.py

# https://leetcode.com/problems/reverse-degree-of-a-string/description/


class Solution:
    def reverseDegree(self, s: str) -> int:
        result = 0
        for i,j in enumerate(s, start=1):
            position_char = ord(j)-96
            reversal = 27 - position_char
            result += (reversal*i)

        return result